task-canvas/script.js

```diff
@@ -0,0 +1,472 @@
+                shapes.push(shape);
+                break;
+            case "Rectangle":
+                shape = new Rectangle(ctx, item.color, item.coords);
+                shapes.push(shape);
+                break;
+            case "Circle":
+                shape = new Circle(ctx, item.color, item.coords);
+                shapes.push(shape);
+                break;
+            }
+        });
+        shapes.forEach(item => item.render(ctx));
+    }
+
+    setMovingShape(shape) { 
+        if (shape) {
+            this.movingShape = shape;
+        } else {
+            this.movingShape = null;
+        }
+    }
+
+    removeAll = () => {
+        const answer = confirm("Your creating will be removed, are you sure?"); 
+        if (answer) {
+            this.arrayOfShapes.splice(0, this.arrayOfShapes.length);
+            this.clear();
+        }
+    }
+
+    newColor(event) {
+        this.color = event.target.value;
+        let pallete = event.target.closest('.label-colors');
+        pallete.style.backgroundColor = `${this.color}`;
+    }
+
+    watchColorPicker(event) {
+        this.color = event.target.value;
+        if (this.getControlType() === "new"){
+            this.movingShape.setColor(this.color);
+        }
+    }
+
+    finishShape = () => {
+        let shape;
+        this.canvas.removeEventListener('mousemove', this.move);
+        switch (this.getControlType()) {
+            case "new":
+                break;
+            case "pencil":
+                this.clear();
+                shape = new BrokenLine(this.ctx, this.color, this.amountCoords);
+                this.arrayOfShapes.push(shape);
+                break;
+            case "rectangle":
+                this.clear();
+                shape = new Rectangle(this.ctx, this.color, this.coords.x, this.coords.y, this.coords.vx, this.coords.vy);
+                this.arrayOfShapes.push(shape);
+                break;
+            case "circle":
+                this.clear();
+                shape = new Circle(this.ctx, this.color, this.coords.x, this.coords.y, this.coords.vx, this.coords.vy);
+                this.arrayOfShapes.push(shape);
+                break;
+        }
+        this.arrayOfShapes.forEach(item => item.render(this.ctx));
+    }
+
+    drawShape(x, y, vx, vy, controlType, color) {
+        const ctx = this.ctx;
+        let shape;
+        switch (controlType) {
+            case "new":
+                let deltaX = this.coords.vx - this.coords.x;
+                let deltaY = this.coords.vy - this.coords.y;
+                shape = this.movingShape;
+
+                if (shape) {
+                    shape.moveShape(deltaX, deltaY);
+                    this.coords.x = this.coords.vx;
+                    this.coords.y = this.coords.vy;
+                    this.clear();
+                }
+                break;
+            case "pencil":
+                shape = new BrokenLine(ctx, color, x, y, vx, vy);
```

Comment:
Do you really need `shape = ` part here?
(I don't see where you're using `shape` below)